keps/sig-scheduling/5234-dra-resourceslice-mixins/README.md

```diff
@@ -0,0 +1,853 @@
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [x] (R) Production readiness review completed
+- [x] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+With Dynamic Resource Allocation (DRA), DRA drivers publish information about
+the devices that they manage in ResourceSlices. This information is used by the
+scheduler when selecting devices for user requests in ResourceClaims.
+
+With this KEP, DRA drivers can define metadata in mixins separately from specific
+devices and include them in a device by reference. This reduces the duplication
+in ResourceSlices and allows for more compact device definitions. 
+
+## Motivation
+
+DRA requires that drivers publish all available devices on a node/cluster in
+`ResourceSlice` objects. There are scenarios where the number of devices
+can be pretty large and each device might have a relatively large amount
+of metadata associated with it, primarily in the form of attributes and
+capacity. This has a few consequences:
+
+* Several of the devices might have similar metadata, resulting in a lot of
+  duplication between the published devices.
+* The size of the data required to specify each device reduces the number of
+  devices that can be defined in a single ResourceSlice without hitting the
+  Kubernetes size limit.
+
+The latter can be addressed by splitting the devices across multiple
+`ResourceSlice`s within a single pool, but that isn't always an option.
+In particular, DRA currently doesn't allow sharing counters across `ResourceSlice`s,
+meaning that the number of devices that can fit into a single `ResourceSlice`
+also limits the number of partitionable devices for a single physical device.
+
+### Goals
+
+- Enable a more compact way to define devices in ResourceSlices so duplication can
+  be reduced and a larger number of devices can be published within a single
+  ResourceSlice.
```

Comment:
Second goal:
- Enable defining devices with more attributes and capacities and counter sets with more counters.

Or is that just a side effect, not a goal?